lib/ronin/cli/commands/ip.rb

```diff
@@ -226,18 +239,32 @@ def process_value(ip)
         #
         def format_ip(ip)
           if options[:hex]
-            "0x%x" % ip.to_i
+            "0x%x" % ip.to_i 
+          elsif options[:hex_octet]
+            octets = ip.to_s.split(".").map!(&:to_i)
+            octets.map { |octet| octet.to_s(16) }.join(".")
           elsif options[:decimal]
             "%u" % ip.to_i
           elsif options[:octal]
-            "0%o" % ip.to_i
+            to_octal(ip.to_i)
+          elsif options[:octal_octet]
+            octets = ip.to_s.split(".").map!(&:to_i)
+            octets.map { |octet| to_octal(octet) }.join(".")
           elsif options[:binary]
             "%b" % ip.to_i
+          elsif options[:ipv6_compat]
+            IPAddr.new(ip.to_s).ipv4_mapped.to_s
           else
             ip.to_s
           end
         end
 
+        private
+
+        def to_octal(n)
+          "0%o" % n
+        end
```

Comment:
`to_octal` can probably be inlined since it's literal just a format string call.

Reply:
Good catch.  This is kind of vestigial.  I started to extract bits from `#format_ip`.  I'll put it back.